bin/specextract

```diff
@@ -828,860 +753,1086 @@ def clip_wmap(wmapfile,threshold,tmpdir):
-        
-            if wmap_clip:
-                clip_wmap(tdetwmap.name,wmap_thresh,tmpdir)
+    with new_pfiles_environment(ardlib=False,copyuser=False):
+        dmgroup.punlearn()
+
+        dmgroup.infile = f"{phafile}[SPECTRUM]"
+        dmgroup.outfile = grpout
+        dmgroup.binspec = spec
+        dmgroup.grouptype = gtype
+        dmgroup.grouptypeval = val
+        dmgroup.ycolumn = "counts"
+        dmgroup.xcolumn = "channel"
+        dmgroup.tabcolumn = ""
+        dmgroup.clobber = clobber
+        dmgroup.verbose = verbose
+
+        dmgroup()
+    
+    return grpout
 
-            #++++++++++++++++++++++++++++++++++++++++++
 
-        except (OSError,IOError):
-            raise IOError("Error generating WMAP w/sky2tdet!")
-			
-        try:
-            # ARF output file
-            arffile = full_outroot + ".arf"
-            
-            # output weight file used in mkrmf
-            weightfile = tempfile.NamedTemporaryFile(suffix=".wfef",dir=tmpdir)
-            
-            mkwarf.punlearn()
 
-            mkwarf.infile = f"{tdetwmap.name}[wmap]"
-            mkwarf.outfile = arffile
-            mkwarf.weightfile = weightfile.name
-            mkwarf.mskfile = mskfile
-            mkwarf.dafile = dafile
-            mkwarf.spectrumfile = ""
-            mkwarf.egridspec = ebin
-            mkwarf.detsubsysmod = "BPMASK=0x03ffff"
-            mkwarf.clobber = "yes"
-            mkwarf.verbose = str(verbose)
+def edit_headers(verbose, infile, key, val, *args, **kwargs):
+    """
+    Update or add the infile header keyword with a specified value.
+    """
+    
+    unit = kwargs.get("unit",None) # optional argument
+    comment = kwargs.get("comment",None)
 
-            mkwarf()
-			
-        except (OSError,IOError):
-            raise IOError("Error generating weighted ARF w/mkwarf!")
-			
-    except IOError as err_msg:
-        print(err_msg)
+    with new_pfiles_environment(ardlib=False,copyuser=False):
+        dmhedit.punlearn()
 
-        raise IOError(f"Failure to create weighted ARF.  Possible causes include: zero counts in the input region; a memory allocation error; or corrupt ARDLIB.  Try running {toolname} with weight=no, binarfmap!=1, or punlearn ardlib.")
+        dmhedit.infile = infile
+        dmhedit.filelist = "none"
+        dmhedit.operation = "add"
+        dmhedit.key = key
+        dmhedit.value = str(val)
+        dmhedit.verbose = str(verbose)
 
-    return arffile, weightfile.name, weightfile, tdetwmap
+        if unit is not None:
+            dmhedit.unit  = str(unit)
 
+        if comment is not None:
+            dmhedit.comment = str(comment)
 
-###############################################################
-# Usage:
-#  check_event_stats(file)
-#
-#
-# Aim:
-#     Use dmlist to determine the event counts
-#     and appropriately error out or proceed with the script
-#
-################################################################
+    return dmhedit()
```

Comment:
With the code as written you've gone to the effort to setting up a temporary pfiles environment but then it gets replaced before you run the dmhedit call. So either

```
        # this is within the new_pfiles_environment context manager
        result = dmhedit()

    return result
```

or just don't bother with `new_pfiles_environment` here (it's not obvious to me why you would need it).

Actually, is there any place that you use the return value from `edit_headers`?

Reply:
I've modified the function, since the returned value is never used, and sensibly you're correct there's no point to using a ```new_pfiles_environment``` here, since I can't see where there's a possibility of a race condition in its usage.